### ISSUE TITLE
paper-dialog-scrollable-1.1.2 moved from flex to fit layout breaking some pages

### ISSUE THREAD

**Issue by ikelos** [PolymerElements/paper-dialog-scrollable#47]
I've only been using polymer for a little while, but I had a component that offered a fixed height list down one side of a larger scrollable pane.  It was convenient to put the list in its own scrollable container (since it can sometimes be significantly longer than the page it's attached to), and up until version 1.1.2, it appeared that paper-dialog-scrollable did not require a paper-dialog (see issue #15).

Having upgraded to 1.1.2 (and 1.1.3 and 1.1.4) all the pages using that component broke, because the paper-scrollable-dialog has a height of 0 and was not inside anything implementing the paper-dialog behaviour.  Examining the code, this now seems to force a fit layout, rather than the previous flex layout, and I have yet to find a way to gracefully override that from outside the component.  One workaround would require housing it within a relative, fixed-size component, but this will then sometimes be too large (when the list is empty) or too small (when the list reaches the scroll limit).

As a standalone component it was very useful to be able to have an area that could grow to a maximum limit and then start scrolling with all the paper niceties.  It seems as though this could be averted by not including the "else if" part of the condition in _ensureTarget.  This would still allow a user to manually apply the fit layout when required, but also allow them to change the layout as they required.

If such a change isn't desired or feasible, then the documentation for using this outside of a dialog should really be updated to clarify that it now must be used within a relative-position, fixed-size element (much like the flex-layout documentation points out).

**Comment by valdrinkoshi** [comment]
Hi @ikelos, `paper-dialog-scrollable` sets the `fit` class to `#scrollable`, not to the host, which is needed to have the `#scrollable` take the full width/height of its parent (the host, `paper-dialog-scrollable`). When this element is used inside a `paper-dialog`, we don't need to set this class, as it will be the sizing target, hence it will be used to calculate the max-width/max-height.
We use `fit` instead of `flex` because the latter causes rendering problems in IE11 (see [this issue](https://github.com/PolymerElements/paper-dialog-scrollable/issues/40))

If you want to use this outside a `paper-dialog`, you can still do it, just make sure you give a size to `paper-dialog-scrollable`, or make it `flex` (as it's done in the [demo page](https://github.com/PolymerElements/paper-dialog-scrollable/blob/master/demo/index.html#L45)).

Hope it helps, closing this for now. Will reopen if you think the problem persists :)

**Comment by ikelos** [comment]
Yep, that's perfect thanks!  I was misusing flex (not in a vertical layout, so the expansion wasn't happening as I expected).  It is still a slightly unexpected change to existing components, but given that the workaround is reasonably trivial I'm not too concerned.  Thanks again for the quick response!  5:)

**Comment by valdrinkoshi** [comment]
No problem, happy to help!
Thinking more of this, probably we could better handle the case when `paper-dialog-scrollable` is not contained in a `paper-dialog`, but the name already suggests its usage and the plan is to move this element into `paper-dialog`.
I'll reopen this one anyways, and assign a lower priority ;)

**Comment by valdrinkoshi** [comment]
@ikelos if you feel like it, feel free to propose a fix for this, here's our [contributing guidelines](https://github.com/PolymerElements/paper-dialog-scrollable/blob/master/CONTRIBUTING.md).

**Comment by ikelos** [comment]
Given there was reason behind the layout change and it was simply me misusing some of the other components that caused me problems, I'm not sure I'm best placed to contribute back a reasonable fix!  5:)

Given its utility, it seems like a shame to hide it away in the paper-dialog component though.  It appears like it might be well suited to cards as well as dialogs (although I'm not sure if fits in with the design philosophy of paper-card)...

**Comment by Lange** [comment]
Apologies for bumping this issue that is more than a year old, but I wanted to add my two cents that I find `paper-dialog-scrollable` to be a very useful element, and I am frequently using it outside of `paper-dialog` elements. The main thing that I like about it are the lines that appear at the top and bottom when there is more content to scroll to in that direction. If there were another element that implemented just that, I would use it.
